### PR TITLE
Add info commandstats metrics for redis

### PR DESCRIPTION
**Pull request by ptqa**
I've added metrics for "info commandstats" output. I'm new to Go, so not sure if this code is ok, but it works.

### CODE REVIEW

File: cmd/scollector/collectors/redis_unix.go [hunk 1]
```diff
@@ -221,8 +233,9 @@ func c_redis() (opentsdb.MultiDataPoint, error) {
 				continue
 			}
 			sp := strings.Split(line, ":")
+			match, _ := regexp.MatchString("cmdstat_([a-z]+)",sp[0])
 			if len(sp) < 2 || !redisFields[sp[0]] {
-				continue
+				if !match {continue}
```
Comment (captncraig) [inline cmd/scollector/collectors/redis_unix.go]: why not move the `if match` block above this to remove the nested if stmt.

Comment (captncraig) [inline cmd/scollector/collectors/redis_unix.go]: This file also needs to be run through `go fmt`

File: cmd/scollector/collectors/redis_unix.go [hunk 2]
```diff
@@ -236,6 +249,16 @@ func c_redis() (opentsdb.MultiDataPoint, error) {
 				Add(&md, "redis."+sp[0], status(sp[1]), tags, metadata.Unknown, metadata.None, "")
 				continue
 			}
+			if match {
+				cmdstats := strings.Split(sp[1],",")
+				cmdstats_calls := strings.Split(cmdstats[0],"=")
```
Comment (captncraig) [inline cmd/scollector/collectors/redis_unix.go]: We need bounds checking any time we start indexing into split strings.

File: cmd/scollector/collectors/redis_unix.go [hunk 3]
```diff
@@ -182,13 +180,22 @@ func c_redis() (opentsdb.MultiDataPoint, error) {
...
+		}
+		info, info_err := c.Do("info")
+		if info_err != nil {
+			Error = info_err
 			continue
 		}
+
 		tags := instance.Copy()
-		infoSplit := strings.Split(string(lines.([]uint8)), "\n")
+		combined_lines := string(info.([]uint8)) + string(info_command.([]uint8))
```
Comment (captncraig) [inline cmd/scollector/collectors/redis_unix.go]: Why combine at all? Why not iterate over info_command separately? It seems like a lot of the conditional logic in later stages would be simpler by not combining.

**Comment by ptqa** [comment]
If I leave a comment may be it's possible to reopen...

**Pull request closed without merging**